ibm/data_source_ibm_iam_role_actions.go

```diff
@@ -0,0 +1,73 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceIBMIAMRoleAction() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceIBMIAMRoleActionRead,
+
+		Schema: map[string]*schema.Schema{
+			"service": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Service Name",
+				ForceNew:    true,
+			},
+			"reader": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Reader action ids",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"manager": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "manager action ids",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"reader_plus": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "readerplus action ids",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"writer": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "writer action ids",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+
+}
+
+func datasourceIBMIAMRoleActionRead(d *schema.ResourceData, meta interface{}) error {
+	iampapv2Client, err := meta.(ClientSession).IAMPAPAPIV2()
+	if err != nil {
+		return err
+	}
+
+	serviceName := d.Get("service").(string)
+	d.SetId(serviceName)
+	serviceRoles, err := iampapv2Client.IAMRoles().ListServiceRoles(serviceName)
+	if err != nil && !strings.Contains(err.Error(), "404") {
```

Comment:
No need to check all strings.Contains(err.Error(), "404" since its datasource just give back wt err it gives back